### PR TITLE
Add schedules to android app 

### PR DESCRIPTION
**Pull request by patrickkenzie**
This adds zuluru schedule scraping for the backend, and support in the android app to set up games based on the schedule. Stat keepers will now just have to pick the game slot, with the teams pre-selected for them!

The android app still has the team select as an option if the schedule ends up wrong or missing. It will auto fallback to the team select if the server doesn't send any games.

### CODE REVIEW

**Comment by patrickkenzie** [comment]
We'll also need to set up a job to run the schedule-sync cli command, probably the same as whatever we're doing for roster sync!

File: server/models/matchup.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from .db import db
+from .league import League
+
+class Matchup(db.Model):
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    league_id = db.Column(db.Integer, db.ForeignKey('league.id'), nullable=False)
+    home_team_id = db.Column(db.Integer, db.ForeignKey('team.id'), nullable=False)
+    away_team_id = db.Column(db.Integer, db.ForeignKey('team.id'), nullable=False)
+    week = db.Column(db.Integer)
+    game = db.Column(db.Integer)
```
[the remark this answers is not in the document]
Comment (patrickkenzie) [reply server/models/matchup.py]: :+1:

File: server/lib/schedule_sync.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    def update_schedule(self):
+        db.session.query(Matchup).filter_by(league_id = self.league_id).delete()
+
+        matchups = self.load_schedule()
+        print (len(matchups), "games retrieved")
+
+        for matchup in matchups:
+            db.session.add(matchup)
+
+        db.session.commit()
```
[the remark this answers is not in the document]
Comment (patrickkenzie) [reply server/lib/schedule_sync.py]: Good call! They ended up with a lot of similar code, I merged them. Left the cli command for separate schedule syncing as it's handy for testing at the moment.

File: android/app/src/main/java/org/ocua/parity/model/Matchup.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package org.ocua.parity.model;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+
+public class Matchup {
+    public final int week;
+    public final int gameSlot;
```
[the remark this answers is not in the document]
Comment (patrickkenzie) [reply android/app/src/main/java/org/ocua/parity/model/Matchup.java]: Yeah, I figured that would be next quality of life update too! Shouldn't be too hard to parse that out of the schedule page.

**Comment by patrickkenzie** [comment]
> My initial reaction to the Matchup name was that we might want to reserve that for player matchups but we have no plans for that I really can't come up with anything better. Game is pretty overloaded. If and when we come up with something better we can change it.

Yeah, that's a good thought for using 'matchup' - pretty much everything else I thought of was some version of 'game'. Maybe something in the area of 'timeslot' could work?

File: server/lib/schedule_sync.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevinhughes27) [reply server/lib/schedule_sync.py]: sweet. It makes sense to have it's own cli command. That is what I will set the heroku schedule to run. Once a day should be enough right?

**Comment by kevinhughes27** [comment]
I only really came up with ScheduledGame or ScheduleItem. I think it is fine to leave it now and see if any names inspire us next time we add anything nearby or if we actually want the matup name back.

File: server/lib/schedule_sync.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrickkenzie) [reply server/lib/schedule_sync.py]: Yeah, that should be fine!

File: android/app/src/main/java/org/ocua/parity/model/Matchup.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickkenzie) [reply android/app/src/main/java/org/ocua/parity/model/Matchup.java]: Ended up just doing this after all. It also meant I could remove the `game_slot` field! We can always add it back if we end up needing it for ordering the game properly when they have the same start times, but I don't think that's a big deal (and we can probably just get away with using the autoincrement IDs for that if need be).

File: server/config.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import os
 
 def development_database_path():
-    use_postgres = True
+    use_postgres = False
```
[the remark this answers is not in the document]
Comment (kevinhughes27) [reply server/config.py]: hmm I take that back. I just rebuilt the full db in #325 and it felt quite a bit quicker than sqlite. It's low effort to support both.

File: server/lib/schedule_sync.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevinhughes27) [reply server/lib/schedule_sync.py]: actually I see you've got it in the zuluru-sync command which means I won't even have to do anything it will just start syncing. It runs in the mornings ~9 I think

File: android/app/build.gradle [hunk 5]
```diff
@@ -51,11 +52,11 @@ android {
             def baseUrl = "https://parity-server.herokuapp.com"
 
             buildConfigField "String", "UPLOAD_URL", "\"" + baseUrl + "/submit_game\""
-            buildConfigField "String", "TEAMS_URL", "\"" + baseUrl + "/api/%s/teams\""
+            buildConfigField "String", "SCHEDULE_URL", "\"" + baseUrl + "/api/%s/schedule\""
 
             dimension = "parity"
         }
-    }
+    submit_game}
```
Comment (kevinhughes27) [inline android/app/build.gradle]: I think this snuck in here. :fire:

**Comment by kevinhughes27** [comment]
:rocket:
